### PR TITLE
Add option for administrator to reset user 2FA

### PR DESCRIPTION
**Pull request by 6543**
by checking the "Reset 2FA" checkbox the 2fa is reset if exist ...

### CODE REVIEW

**Comment by CirnoT** [comment]
I don't think this removes U2F registrations, leaving 2FA in a rather inconsistent state where U2F tokens are registered but 2FA itself is disabled.

**Comment by 6543** [comment]
@CirnoT this is the way it is done ... :

https://github.com/go-gitea/gitea/blob/3d38ecb1fc4ecb1abcfce960d6630fab25497914/routers/user/setting/security_twofa.go#L55-L82

**Comment by CirnoT** [comment]
I still really think U2F registrations should be removed when 2FA is disabled administratively.

On the other hand, maybe they can be also used for other things later on. Won't block because of that.

File: templates/admin/user/edit.tmpl [hunk 1]
```diff
@@ -109,6 +109,12 @@
 					</div>
 				</div>
 				{{end}}
+				<div class="inline field">
```
Comment (CirnoT) [inline templates/admin/user/edit.tmpl:115]: I'd place some divider here, to indicate that it is different UI element from rest of checkboxes.

Comment (CirnoT) [reply templates/admin/user/edit.tmpl:115]: Also I wonder if we could just not display this if user is not 2FA enrolled, we already fetch the state for a list so it should be simple check in template.

Comment (6543) [reply templates/admin/user/edit.tmpl:115]: no it's not a simple template change ... but I'll add the check

**Comment by 6543** [comment]
@CirnoT done -> 42e7d84

**Review by CirnoT — APPROVED** [review]

File: routers/admin/users.go [hunk 2]
```diff
@@ -183,6 +183,16 @@ func prepareUserInfo(ctx *context.Context) *models.User {
...
 	ctx.Data["Sources"] = sources
 
+	ctx.Data["TwoFactorEnabled"] = true
+	_, err = models.GetTwoFactorByUID(u.ID)
+	if err != nil {
+		if !models.IsErrTwoFactorNotEnrolled(err) {
+			ctx.InternalServerError(err)
+			return nil
+		}
+		ctx.Data["TwoFactorEnabled"] = false
```
Comment (lunny) [inline routers/admin/users.go:193]: ~log the error is better so that we know what happened.~

**Review by lunny — APPROVED** [review]

**Pull request merged**